File: Puntonet/Puntonet.Web/Modules/Parameters/Accesories/AccesoriesRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace Puntonet.Parameters
{
    [ConnectionKey("Default"), Module("Parameters"), TableName("[dbo].[Accesories]")]
    [DisplayName("Accesories"), InstanceName("Accesories")]
    [ReadPermission("Administration:General")]
    [ModifyPermission("Administration:General")]
    [LookupScript("Parameters.Accesories", Permission = "*")]
    public sealed class AccesoriesRow : Row<AccesoriesRow.RowFields>, IIdRow, INameRow
    {
        [DisplayName("Id Accesory"), Identity, IdProperty]
        public int? IdAccesory
        {
            get => fields.IdAccesory[this];
            set => fields.IdAccesory[this] = value;
        }

        [DisplayName("Description"), Size(200), NotNull, QuickSearch, NameProperty]
        public string Description
        {
            get => fields.Description[this];
            set => fields.Description[this] = value;
        }

        public AccesoriesRow()
            : base()
        {
        }

        public AccesoriesRow(RowFields fields)
            : base(fields)
        {
        }

        public class RowFields : RowFieldsBase
        {
            public Int32Field IdAccesory;
            public StringField Description;
        }
    }
}

File: Puntonet/Puntonet.Web/Modules/ServerTypes/Parameters/AccesoriesRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface AccesoriesRow {
    IdAccesory?: number;
    Description?: string;
}

export abstract class AccesoriesRow {
    static readonly idProperty = 'IdAccesory';
    static readonly nameProperty = 'Description';
    static readonly localTextPrefix = 'Parameters.Accesories';
    static readonly lookupKey = 'Parameters.Accesories';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<AccesoriesRow>('Parameters.Accesories') }
    static async getLookupAsync() { return getLookupAsync<AccesoriesRow>('Parameters.Accesories') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<AccesoriesRow>();
}
